apps/mark-scan/frontend/src/app_root.tsx

```diff
@@ -564,7 +553,7 @@ export function AppRoot({
         <InsertCardScreen
           appPrecinct={precinctSelection}
           electionDefinition={optionalElectionDefinition}
-          showNoChargerAttachedWarning={!computer.batteryIsCharging}
+          showNoChargerAttachedWarning={!!battery && battery.discharging}
```

Comment:
Is battery every not defined if the query succeeds? Should we `assertDefined(battery)` after the check for `batteryQuery.isSuccess`?

Reply:
It's possible that battery is not defined. In fact, to your main question, it's probably not defined for VxMarkScan. We check battery status by polling the file at `/sys/class/power_supply/BAT0/uevent` or the equivalent with `BAT1`. That file may not exist for a number of reasons. 

I didn't think about the particulars of the VxMarkScan hardware. I'm curious what you'd find if you checked for those files, but I'm guessing they don't exist. Does the VxMarkScan have an onboard UPS? Or are we planning on using an external one? In either case, we probably don't have access to battery level.

If you unplug the VxMarkScan, does it just turn off?

Reply:
Summarizing offline discussion - there is no onboard battery. Power requirements are in flux right now but we might want a data connection from a UPS, so we’re going to leave this in to avoid unnecessary work of removing/adding later.